File: FINHUB-TN-main/src/main/java/tn/finhub/model/SimulatedTrade.java
package tn.finhub.model;

import java.math.BigDecimal;
import java.sql.Timestamp;

public class SimulatedTrade {
    private int id;
    private int userId;
    private String assetSymbol;
    private String action; // BUY or SELL
    private BigDecimal quantity;
    private BigDecimal priceAtTransaction;
    private BigDecimal totalCost;
    private Timestamp transactionDate;

    public SimulatedTrade(int id, int userId, String assetSymbol, String action, BigDecimal quantity,
            BigDecimal priceAtTransaction, BigDecimal totalCost, Timestamp transactionDate) {
        this.id = id;
        this.userId = userId;
        this.assetSymbol = assetSymbol;
        this.action = action;
        this.quantity = quantity;
        this.priceAtTransaction = priceAtTransaction;
        this.totalCost = totalCost;
        this.transactionDate = transactionDate;
    }

    public SimulatedTrade() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getUserId() {
        return userId;
    }

    public void setUserId(int userId) {
        this.userId = userId;
    }

    public String getAssetSymbol() {
        return assetSymbol;
    }

    public void setAssetSymbol(String assetSymbol) {
        this.assetSymbol = assetSymbol;
    }

    public String getAction() {
        return action;
    }

    public void setAction(String action) {
        this.action = action;
    }

    public BigDecimal getQuantity() {
        return quantity;
    }

    public void setQuantity(BigDecimal quantity) {
        this.quantity = quantity;
    }

    public BigDecimal getPriceAtTransaction() {
        return priceAtTransaction;
    }

    public void setPriceAtTransaction(BigDecimal priceAtTransaction) {
        this.priceAtTransaction = priceAtTransaction;
    }

    public BigDecimal getTotalCost() {
        return totalCost;
    }

    public void setTotalCost(BigDecimal totalCost) {
        this.totalCost = totalCost;
    }

    public Timestamp getTransactionDate() {
        return transactionDate;
    }

    public void setTransactionDate(Timestamp transactionDate) {
        this.transactionDate = transactionDate;
    }
}

File: OneDrive/Desktop/FINHUB-WEB-main/src/Entity/SimulatedTrade.php
<?php

namespace App\Entity;

use App\Repository\SimulatedTradeRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: SimulatedTradeRepository::class)]
#[ORM\Table(name: 'simulated_trades')]
class SimulatedTrade
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne]
    #[ORM\JoinColumn(name: 'user_id', referencedColumnName: 'user_id', nullable: false)]
    private ?User $user = null;

    #[ORM\Column(name: 'asset_symbol', length: 20)]
    private ?string $assetSymbol = null;

    #[ORM\Column(name: 'action', length: 10)] // e.g., 'BUY' or 'SELL'
    private ?string $action = null;

    #[ORM\Column(type: Types::DECIMAL, precision: 12, scale: 4)]
    private ?string $quantity = null;

    #[ORM\Column(name: 'price_at_transaction', type: Types::DECIMAL, precision: 12, scale: 4)]
    private ?string $priceAtTransaction = null;

    #[ORM\Column(name: 'total_cost', type: Types::DECIMAL, precision: 12, scale: 4, nullable: true, insertable: false, updatable: false)]
    private ?string $totalCost = null;

    #[ORM\Column(name: 'transaction_date', type: Types::DATETIME_MUTABLE)]
    private ?\DateTimeInterface $transactionDate = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getUser(): ?User
    {
        return $this->user;
    }

    public function setUser(?User $user): static
    {
        $this->user = $user;

        return $this;
    }

    public function getAssetSymbol(): ?string
    {
        return $this->assetSymbol;
    }

    public function setAssetSymbol(string $assetSymbol): static
    {
        $this->assetSymbol = $assetSymbol;

        return $this;
    }

    public function getAction(): ?string
    {
        return $this->action;
    }

    public function setAction(string $action): static
    {
        $this->action = $action;

        return $this;
    }

    public function getQuantity(): ?string
    {
        return $this->quantity;
    }

    public function setQuantity(string $quantity): static
    {
        $this->quantity = $quantity;

        return $this;
    }

    public function getPriceAtTransaction(): ?string
    {
        return $this->priceAtTransaction;
    }

    public function setPriceAtTransaction(string $priceAtTransaction): static
    {
        $this->priceAtTransaction = $priceAtTransaction;

        return $this;
    }

    public function getTotalCost(): ?string
    {
        return $this->totalCost;
    }

    public function setTotalCost(?string $totalCost): static
    {
        $this->totalCost = $totalCost;

        return $this;
    }

    public function getTransactionDate(): ?\DateTimeInterface
    {
        return $this->transactionDate;
    }

    public function setTransactionDate(\DateTimeInterface $transactionDate): static
    {
        $this->transactionDate = $transactionDate;

        return $this;
    }
}